### DISCUSSION TITLE
How to use ENV variables in config

### DISCUSSION THREAD

**Discussion by bab014** [starship/starship#5794] [Q&A]
I hope this is just an easy answer and I'm just not smart enough to realize how to do it. What I would like to do is use environment variables within my starship.toml config. Specifically I have a program that sets some color variables for the terminal, I'd like to be able to grab those within the config and use them throughout.

Is that something that I can accomplish?

**Comment by GANESH-NADKARNI** [comment]
Starship's starship.toml doesn't support arbitrary env-var interpolation directly inside format/style strings — TOML values there are static. But there are two ways to get env var content into your prompt, depending on what you need:

**1. To just display an env var's value** — use the built-in `env_var` module:

```toml
[env_var.MY_COLOR_VAR]
format = "[$env_value]($style)"
```

This shows the variable's content but won't let you use its value *as* a style/color dynamically.

**2. To use the env var's value as an actual color/style** (which sounds like what you want) — use a `custom` module instead, since custom modules can run a shell command and their output is evaluated at prompt render time:

```toml
[custom.mycolor]
command = "echo -n \"$MY_COLOR_VAR\""
when = true
format = "[$output]($output)"
```

Since `command` runs in a real shell, it has full access to your exported env vars, and you can pipe/format the output however you need before Starship renders it. Add `$custom` (or `$custom.mycolor` for just this one) into your top-level `format` string to place it.

If you only need this for a few known vars, `env_var` is simpler; if you need the value to actually change styling dynamically, `custom` is the way to go.

Docs: https://starship.rs/config/#environment-variable and https://starship.rs/config/#custom-commands
